### PR TITLE
Validate credentials on login

### PR DESCRIPTION
**Pull request by thatplguy**
Changes:
 - `akita login` now tests credentials before reporting success
 - `akita login --check` is a new flag that tests existing credentials

### CODE REVIEW

File: cmd/internal/login/login.go [hunk 1]
```diff
@@ -24,28 +29,84 @@ API key information will be stored in ` + cfg.GetCredentialsConfigPath(),
...
-			{
-				Name:   "APIKeyID",
-				Prompt: &survey.Input{Message: "API Key ID:"},
-			},
-			{
-				Name:   "APIKeySecret",
-				Prompt: &survey.Password{Message: "API Key Secret:"},
-			},
-		}
+		if !checkCreds {
```
Comment (liujed) [inline cmd/internal/login/login.go:32]: Nice to have: if credentials are being provided in the environment, warn the user that the credentials they are about to enter will have no effect.

**Review by liujed — APPROVED** [review]

File: cmd/internal/login/flags.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+func registerRequiredFlags() {
+}
+
+func registerOptionalFlags() {
+	Cmd.Flags().BoolVar(
+		&checkCreds,
+		"check",
+		false,
+		"If true, checks authentication with existing credentials rather than asking for new credentials.",
```
[the remark this answers is not in the document]
Comment (thatplguy) [reply cmd/internal/login/flags.go]: Good point.  I made the message more imperative.  I also changed the format (capitalization, punctuation) to match the other flag help messages, which are lowercase.

Changed in a31c14f.

File: cmd/internal/login/flags.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	)
+
+	Cmd.Flags().BoolVar(
+		&showCreds,
+		"show-creds",
+		false,
+		"If true, prints credentials to stdout.  Useful for debugging.",
```
[the remark this answers is not in the document]
Comment (thatplguy) [reply cmd/internal/login/flags.go]: Ditto -- fixed in a31c14f.

File: cmd/internal/login/login.go (hunk 1 above)
Comment (thatplguy) [reply cmd/internal/login/login.go]: Added in 785f96c.

**Pull request merged**